### ISSUE TITLE
Production issue with Apple Pay

### ISSUE THREAD

**Issue by ScreamZ** [stripe/stripe-react-native#1377]
Hello,

One year ago I made a thread having an issue with Apple Pay: https://github.com/stripe/stripe-react-native/issues/945. One year later I'm working back on that project and I'm trying to fix things but nothing seems to have changed.

# A bit of context

My customer is "renting" power banks, for free in fact, but I need to take a Hold of 35€ in case people don't bring back the power bank after two days to the station.

I followed the doc using `initPaymentSheet` from https://stripe.com/docs/payments/accept-a-payment?platform=react-native&ui=payment-sheet#react-native-apple-pay.

## Google pay

For Google Pay, everything works so far. 

## Apple pay
- I've no issue in staging (com.stride.power.staging), the validation seems to pass.
- But in production, things get worse… And this is really annoying because we are losing a lot of user conversions due to this case.

The only difference is the PUBLISHABLE_KEY (pk_test vs real one) and the bundle identifier (between apps). **I've shared a lot of screenshots with the configuration and code below.**

Here is the log from Stripe, I've no idea what the CVC has to do with apple pay, this is not asked anywhere. (https://dashboard.stripe.com/logs/req_fMMpcFFZtn0oFh)

![Screenshot 2023-05-03 at 19 04 22](https://user-images.githubusercontent.com/6640835/235988386-c6c0c87e-23cb-41a3-89d1-a813bc3ef568.png)

![IMG_A6317638AC18-1](https://user-images.githubusercontent.com/6640835/235985502-8b4ce9db-6f78-411b-89cd-b8b7799f3b47.jpeg)

Thanks !

### Side note
- I'm using two app IDs: `com.stride.power.staging` (works with that config) and `com.stride.power` (production, doesn't work) 
- ✅ I've been able to use Apple Pay without any issue from the Stripe customer portal (now it's proposed, but some months ago it wasn't). So I guess it's related to some configuration. This can be a workaround for waiting for a fix of my issue

# Context
- App link (if you want to try) https://apps.apple.com/fr/app/stride-recharge-%C3%A0-linfini/id1596882572
- OS: iOS
- "@stripe/stripe-react-native": "0.23.3",

## Additional screenshots
![IMG_1493](https://user-images.githubusercontent.com/6640835/235987184-75e5d448-5ec8-4dc3-b957-21a0013ae5d1.PNG)
![IMG_1494](https://user-images.githubusercontent.com/6640835/235987210-cb682c7d-5452-4398-9405-247395df0368.PNG)
![IMG_1495](https://user-images.githubusercontent.com/6640835/235987223-65ee95f8-15e0-48e9-9321-debce5b0f013.PNG)
![Screenshot 2023-05-03 at 19 09 50](https://user-images.githubusercontent.com/6640835/235989938-c505ecfa-2943-4752-bae1-d4b7a01157a0.png)
![Screenshot 2023-05-03 at 19 10 19](https://user-images.githubusercontent.com/6640835/235990041-8e84d089-bc6c-442a-90b4-79648b1c03e5.png)
![Screenshot 2023-05-03 at 19 11 06](https://user-images.githubusercontent.com/6640835/235990207-4bfd2f81-131c-4659-a4df-42f9f2c646f4.png)
![Screenshot 2023-05-03 at 19 11 14](https://user-images.githubusercontent.com/6640835/235990234-32e85689-5d22-4bf4-9f56-5401faf0afac.png)
![Screenshot 2023-05-03 at 19 12 07](https://user-images.githubusercontent.com/6640835/235990709-ef855da1-3c1a-4f34-b88c-3b702a1da986.png)
![Screenshot 2023-05-03 at 19 13 00](https://user-images.githubusercontent.com/6640835/235990836-52bbd628-8790-4085-9c83-4e12d1fe0bc5.png)
![Screenshot 2023-05-03 at 19 14 59](https://user-images.githubusercontent.com/6640835/235991250-2059c23f-f1a6-425e-9a89-4bbd7dfc059e.png)

**Comment by charliecruzan-stripe** [comment]
hey @ScreamZ , do you _only_ get this error when attempting to use apple pay? Inputting the card data directly into payment sheet results in a success?

If it's not working in production then I do think there's an actual validation issue happening here...

**Comment by ScreamZ** [comment]
Yes, this works most on time with payment sheet, but Apple Pay fails (and sometime works I don't know why). Invalid CVC they said but it's never prompted and the whole user base is getting this error

**Comment by charliecruzan-stripe** [comment]
Hm, I'm not sure what could be causing that in the Stripe React Native library. Maybe you have some radar rules set that are causing this?

Could you share an example setup intent ID with me that I can look into?

**Comment by ScreamZ** [comment]
> Hm, I'm not sure what could be causing that in the Stripe React Native library. Maybe you have some radar rules set that are causing this?
> 
> Could you share an example setup intent ID with me that I can look into?

Yeah you have one on first screenshot isn't ?

**Comment by ScreamZ** [comment]
I've switched to the customer portal to add card, this way stripe manage the signature and this works on their side

**Comment by charliecruzan-stripe** [comment]
Hey folks, I did some digging and the decline is not from Stripe, but from the card network. It shouldn't have anything to do with your iOS certs either. If you try a non-visa card, does it work?

**Comment by Kai599** [comment]
I don't have access to a non-visa card.

**But**: I've tried two different visa cards, same error (invalid_cvc). Both cards work fine when you pay directly with the card and not trough ApplePay.
Both cards even work with ApplePay when you pay for something (createStripePayment). The payment method is saved afterwards, so I can do paments in the future without getting the Stripe payment sheet. 
The error (invalid_cvc) only appears if you want to add the card as a payment method without paying for something (createSetupIntent).

**Comment by Kai599** [comment]
EDIT: The problem really only exists with visa cards. A mastercard works fine. Thus, the deadly combination is ApplePay + createSetupIntent + Visa card.

**Comment by ScreamZ** [comment]
> I don't have access to a non-visa card.
> 
> **But**: I've tried two different visa cards, same error (invalid_cvc). Both cards work fine when you pay directly with the card and not trough ApplePay. Both cards even work with ApplePay when you pay for something (createStripePayment). The payment method is saved afterwards, so I can do paments in the future without getting the Stripe payment sheet. The error (invalid_cvc) only appears if you want to add the card as a payment method without paying for something (createSetupIntent).

I didn't had this case, i was only setting setupIntent, but sometimes its works, its very random, even on my own visa card, sometime it works, sometime it doesn't. (I've made multiple try)

**Comment by eirikurn** [comment]
Also have this problem. Comparing the UX of the Stripe sheet vs another app where this exact flow works (register visa card as a payment method using Apple Pay "without payment") I see this potential hint.

- Stripe Apple Pay sheet shows "{CompanyName}\nAmount Pending".
- Other app shows "Pay {CompanyName}\n0 {Currency}".

Not sure if that makes any difference behind the scenes, but at least VISA cards seem to work in the other and not Stripe.

**Comment by ScreamZ** [comment]
> Also having this problem. Comparing the UX of the Stripe sheet vs another app where this exact flow works (register visa card as a payment method using Apple Pay "without payment") I see this potential hint:
> 
> * Stripe Apple Pay sheet shows "{CompanyName}\nAmount Pending".
> * Other app shows "Pay {CompanyName}\n0 {Currency}".
> 
> Not sure if that makes any difference behind the scenes, but at least VISA cards seem to work in the other app and not ours. Note, the other app is not using Stripe, or at least not Stripe's payment sheet.
> 
> **EDIT: Found a workaround.** Pass `paymentSummaryItems` (or I guess `cartItems` in the latest version) to `initPaymentSheet` with the "Immediate" paymentType. This works for me:
> 
> ```
> await initPaymentSheet({
>   applePay: {
>     merchantCountryCode,
>     paymentSummaryItems: [
>       {
>         paymentType: 'Immediate',
>         amount: '0',
>         label: 'Some Label',
>         isPending: true,
>       },
>     ],
>   },
> })
> ```
> 
> This shows the same UI as before but you can also skip `isPending` to show a payment of 0. We have yet to deploy this and I don't know if there are side effects of this change.

Stripe is generally not working well with payment intent = 0 because they don't support it as it doesn't allow them to proceed money cut on their purpose. This could be nice if this is working.

**Comment by VTSingle** [comment]
@ScreamZ 
sorry for the of top question, but maybe you can leave some recommendations for me in my question 

https://github.com/stripe/stripe-react-native/issues/1494

Thank you a lot in advance :)

**Comment by hasnainali9** [comment]
Hey i'm same issue with visa card. Here is my params.
<img width="745" alt="image" src="https://github.com/user-attachments/assets/7143b099-08ff-47d2-80f7-448a04e6980c" />
Does anyone able to solve this issue?

**Comment by porter-stripe** [comment]
Hi all, is this still an issue you are experiencing?

**Comment by ScreamZ** [comment]
No idea, it's been almost 3 year that this has been opened, and didn't made any change to the matching application. But you have full reproduction, if you wish to try

**Comment by emdotem** [comment]
I'm having the same issue on web, any clue?

**Comment by nickmanderfield** [comment]
Still an issue, particularly ApplePay + createSetupIntent + Visa card, including with Stripe Link card adding.
